wallet/src/main/scala/org/bitcoins/wallet/models/OutgoingTransactionTable.scala

```diff
@@ -0,0 +1,79 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.crypto._
+import org.bitcoins.core.currency._
+import org.bitcoins.core.protocol.transaction._
+import org.bitcoins.core.wallet.fee.SatoshisPerByte
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.{PrimaryKey, ProvenShape}
+
+/**
+  * Represents a relevant transaction for the wallet that we should be keeping track of
+  * @param txIdBE Transaction ID
+  * @param txFee fee paid by the transaction
+  * @param expectedFeeRate Fee rate the wallet expected to pay
+  * @param actualFeeRate Fee rate the transaction actually paid
+  */
+case class OutgoingTransactionDb(
+    txIdBE: DoubleSha256DigestBE,
+    txFee: CurrencyUnit,
+    expectedFeeRate: SatoshisPerByte,
+    actualFeeRate: SatoshisPerByte) {
+  lazy val txId: DoubleSha256Digest = txIdBE.flip
+}
+
+object OutgoingTransactionDb {
+
+  def fromTransaction(
+      tx: Transaction,
+      expectedFeeRate: SatoshisPerByte,
+      inputAmount: CurrencyUnit): OutgoingTransactionDb = {
+    val feePaid = inputAmount - tx.outputs.map(_.value).sum
+    val feeRate = feePaid.satoshis.toLong / tx.baseSize
```

Comment:
Yikes. Is there no helper methods for calcuations like this? This seems _so_ easy to forget to update in the future when we change the `expectedFeeRate` type.

Reply:
I will implement this in #1236